File: free-platform-book/free-platform-book-api/src/main/java/top/yangwulang/platform/entity/book/BookInfo.java
package top.yangwulang.platform.entity.book;

import io.swagger.v3.oas.annotations.media.Schema;
import org.babyfish.jimmer.Immutable;
import org.babyfish.jimmer.sql.*;

import javax.validation.constraints.NotNull;
import javax.validation.constraints.Null;
import java.util.List;

/**
 * 书籍信息
 */
@Entity
@Table(name = "book_info")
@Schema(description = "书籍信息")
public interface BookInfo {
    /**
     * 小说信息主键
     */
    @Id
    @Schema(description = "书籍信息主键")
    String id();

    /**
     * 小说名称
     */
//    @NotNull
    @Schema(description = "书籍名称")
    @Null String bookName();

    /**
     * 小说作者
     */
    @Schema(description = "书籍作者")
    @Null String author();

    /**
     * 描述
     */
    @Schema(description = "描述")
    @Column(name = "`describe`")
    @Null String describe();

    /**
     * 状态 （连载，完结）
     */
    @Schema(description = "书籍状态")
    @Null String status();

    /**
     * 小说图片地址
     */
    @Schema(description = "书籍地址")
    @Null String bookImgPath();

    /**
     * 小说分类
     */
    @Schema(description = "书籍分类")
    @Null String category();

    /**
     * 小说来源
     */
    @Key
    @Schema(description = "书籍来源")
    @Null String bookFrom();

    @OneToMany(mappedBy = "book")
    @Schema(description = "章节信息")
    List<BookChapter> chapters();
}

File: platform/src/main/kotlin/top/yangwulang/platform/entity/book/BookInfo.kt
package top.yangwulang.platform.entity.book

import lombok.ToString
import org.hibernate.annotations.GenericGenerator
import java.io.Serializable
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id
import javax.persistence.Table

@Entity
@Table(name = "book_info")
@ToString
open class BookInfo : Serializable {
    @Id
    @GeneratedValue(generator = "snowflake")
    @GenericGenerator(name = "snowflake", strategy = "top.yangwulang.platform.utils.SnowflakeGenerator")
    @Column(name = "id", nullable = false, length = 64)
    open var id: String? = null

    @Column(name = "book_name", nullable = false, length = 500)
    open var bookName: String? = null

    @Column(name = "author", length = 500)
    open var author: String? = null

    @Column(name = "`describe`", length = 5000)
    open var describe: String? = null

    @Column(name = "status", length = 50)
    open var status: String? = null

    @Column(name = "book_img_path", length = 500)
    open var bookImgPath: String? = null

    @Column(name = "category", length = 200)
    open var category: String? = null

    @Column(name = "book_from", length = 2000)
    open var bookFrom: String? = null
}